src/vstest.console/CommandLine/Interfaces/IInferHelper.cs

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.VisualStudio.TestPlatform.CommandLine.Processors
+{
+    using System.Collections.Generic;
+    using Microsoft.VisualStudio.TestPlatform.ObjectModel;
+
+    /// <summary>
+    /// Defines interface for infering run configuration values.
+    /// </summary>
+    internal interface IInferHelper
+    {
+        /// <summary>
+        /// Determines Framework from sources.
+        /// </summary>
+        Framework AutoDetectFramework(List<string> sources);
```

Comment:
May be this should be along side other runsettings services. There's also a InferRunSettingsHelper, how do we differentiate behavior of these units? And there's the xmlrunsettingsutilities, runsettingsmanager and runsettingsprovider too.

Reply:
How do we make runsettings management cohesive?